File: buzzer.cpp
#define WATER_SENSOR_PIN A0
#define BUZZER_PIN 9    

const int CRITICAL_LEVEL = 300; // Seuil critique pour le niveau d'eau (à ajuster)
int waterLevel = 0;
bool alertActive = false; // État de l’alerte

void setup() {
  pinMode(WATER_SENSOR_PIN, INPUT);
  pinMode(BUZZER_PIN, OUTPUT);
  Serial.begin(9600); // Pour afficher les données sur le moniteur série
}

void loop() {
  // Lecture du niveau d'eau
  waterLevel = analogRead(WATER_SENSOR_PIN);
  Serial.print("Niveau d'eau : ");
  Serial.println(waterLevel);

  // Cas : seuil critique atteint
  if (waterLevel < CRITICAL_LEVEL && !alertActive) {
    tone(BUZZER_PIN, 2000); // Active buzzer (2 kHz)
    Serial.println("🔴 ALERTE : Niveau d'eau critique !");
    alertActive = true;

  // Cas : retour à la normale
  } else if (waterLevel >= CRITICAL_LEVEL && alertActive) {
    noTone(BUZZER_PIN);     // Désactive buzzer
    Serial.println("✅ INFO : Retour à un niveau normal.");
    alertActive = false;
  }

  delay(600); // Pause pour éviter des lectures trop fréquentes
}

File: buzzer.py
import serial
import time
import RPi.GPIO as GPIO

# --- Configuration ---
CRITICAL_DEPTH = 0.8  # Seuil critique en mètres
BUZZER_PIN = 17       # GPIO du buzzer

# Initialisation GPIO
GPIO.setmode(GPIO.BCM)
GPIO.setup(BUZZER_PIN, GPIO.OUT)
GPIO.output(BUZZER_PIN, GPIO.LOW)

# Lecture de la sonde DST800
def read_depth():
    try:
        with serial.Serial('/dev/ttyUSB1', baudrate=4800, timeout=1) as ser:
            line = ser.readline().decode('ascii', errors='replace')
            if line.startswith('$SDDBT'):
                parts = line.split(',')
                depth = float(parts[1])
                return depth
    except Exception as e:
        print(f"[ERREUR] Lecture de la sonde : {e}")
    return None

def trigger_alert():
    GPIO.output(BUZZER_PIN, GPIO.HIGH)

def stop_alert():
    GPIO.output(BUZZER_PIN, GPIO.LOW)

def main():
    alert_active = False

    try:
        while True:
            depth = read_depth()
            if depth is not None:
                print(f"[INFO] Profondeur détectée : {depth:.2f} m")

                if depth < CRITICAL_DEPTH and not alert_active:
                    print("[ALERTE] Niveau critique atteint ! Déclenchement du buzzer.")
                    trigger_alert()
                    alert_active = True

                elif depth >= CRITICAL_DEPTH and alert_active:
                    print("[INFO] Retour à la normale. Arrêt de l'alerte.")
                    stop_alert()
                    alert_active = False

            else:
                print("[ATTENTION] Donnée non reçue ou invalide.")

            time.sleep(1)

    except KeyboardInterrupt:
        print("\nArrêt du programme par l'utilisateur.")
    finally:
        stop_alert()
        GPIO.cleanup()

if __name__ == "__main__":
    main()
